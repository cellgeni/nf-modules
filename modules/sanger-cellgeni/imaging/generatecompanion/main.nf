process IMAGING_GENERATECOMPANION {
    tag "${meta.id}"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "quay.io/cellgeni/imagetileprocessor:0.1.16"

    input:
    tuple val(meta), val(round), path(image_root_folder)

    output:
    tuple val(meta), path("${out_csv_name}"), emit: csv
    tuple val(meta), path("${out_companion}"), emit: companion
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}_image${round}"
    out_csv_name = "${prefix}_image_tiles.csv"
    out_companion = "${prefix}.companion.ome"
    """
    generate_companion.py run \\
        --image_root_folder ${image_root_folder} \\
        --tiles_csv ${out_csv_name} \\
        --companion_xml ${out_companion} \\
        --prefix ${prefix} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        imagepreprocess: \$(generate_companion.py version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}_image${round}"
    out_csv_name = "${prefix}_image_tiles.csv"
    out_companion = "${prefix}_image.companion.ome"
    """
    touch ${out_csv_name}
    touch ${out_companion}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        imagepreprocess: \$(generate_companion.py version)
    END_VERSIONS
    """
}
