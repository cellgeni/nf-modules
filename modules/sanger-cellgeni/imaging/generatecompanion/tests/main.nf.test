// nf-test test main.nf.test in tests folder
nextflow_process {

    name "Test Process IMAGING_GENERATECOMPANION"
    script "../main.nf"
    process "IMAGING_GENERATECOMPANION"
    config "./nextflow.config"

    tag "modules"
    tag "modules_sanger-cellgeni"
    tag "imaging"
    tag "imaging/generatecompanion"

    test("generatecompanion - PerkinElmer") {

        when {
            params {
                module_args = ""
            }
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    0,
                    file('s3://spatial_demo_datasets/PerkinElmer/Hoechst__2023-03-07T10_17_54-Measurement 2/Images/', checkIfExists: true),
                    'Index.idx.xml'
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("generatecompanion -  stub") {

        options "-stub"

        when {
            params {
                module_args = ""
            }
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    0,
                    file('s3://spatial_demo_datasets/PerkinElmer/Hoechst__2023-03-07T10_17_54-Measurement 2/Images/', checkIfExists: true),
                    'Index.idx.xml'
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
