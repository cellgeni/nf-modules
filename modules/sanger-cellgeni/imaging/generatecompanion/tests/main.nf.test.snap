{
    "generatecompanion - hcs": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image_tiles.csv:md5,d76b6ed00fb1dbef53ae2c5930a52e0e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image_companion.xml:md5,e85270402442371cdbf0efd9524d6fd1"
                    ]
                ],
                "2": [
                    "versions.yml:md5,82df7297bf00de74e36c7db6bfbfa342"
                ],
                "companion": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image_companion.xml:md5,e85270402442371cdbf0efd9524d6fd1"
                    ]
                ],
                "csv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image_tiles.csv:md5,d76b6ed00fb1dbef53ae2c5930a52e0e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,82df7297bf00de74e36c7db6bfbfa342"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-05-16T23:46:38.519945769"
    },
    "generatecompanion - PerkinElmer": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image0_image_tiles.csv:md5,358ff90ea6674c10a433e0262397d059"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image0.companion.ome:md5,286c55aaafb141032753cfe3929157e0"
                    ]
                ],
                "2": [
                    "versions.yml:md5,82df7297bf00de74e36c7db6bfbfa342"
                ],
                "companion": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image0.companion.ome:md5,286c55aaafb141032753cfe3929157e0"
                    ]
                ],
                "csv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image0_image_tiles.csv:md5,358ff90ea6674c10a433e0262397d059"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,82df7297bf00de74e36c7db6bfbfa342"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-08-06T19:49:08.199772238"
    },
    "generatecompanion - wsi": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image_tiles.csv:md5,f7a10853ca180752915d5b4773375169"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image_companion.xml:md5,689ed60d4c30d1b4eeecb8d714231594"
                    ]
                ],
                "2": [
                    "versions.yml:md5,82df7297bf00de74e36c7db6bfbfa342"
                ],
                "companion": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image_companion.xml:md5,689ed60d4c30d1b4eeecb8d714231594"
                    ]
                ],
                "csv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image_tiles.csv:md5,f7a10853ca180752915d5b4773375169"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,82df7297bf00de74e36c7db6bfbfa342"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-05-16T23:47:13.958222538"
    },
    "generatecompanion - hcs - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_image_tiles.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_image_companion.xml:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,c940419a37896e991debf29d01bf57a1"
                ],
                "companion": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_image_companion.xml:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "csv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_image_tiles.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c940419a37896e991debf29d01bf57a1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-28T21:48:28.374592138"
    },
    "generatecompanion -  stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image0_image_tiles.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image0_image.companion.ome:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,82df7297bf00de74e36c7db6bfbfa342"
                ],
                "companion": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image0_image.companion.ome:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "csv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_image0_image_tiles.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,82df7297bf00de74e36c7db6bfbfa342"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-06-10T11:30:34.507793418"
    }
}